cmd/gcp-controller-manager/healthz/healthz_test.go

```diff
@@ -0,0 +1,86 @@
+package healthz
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestHandler(t *testing.T) {
+	passingCheck := func(context.Context) error { return nil }
+	failingCheck := func(context.Context) error { return errors.New("failed") }
+	blockedCheck := func(ctx context.Context) error {
+		<-ctx.Done()
+		return ctx.Err()
+	}
+	tests := []struct {
```

Comment:
Add a test for multiple failures to test the issue I raised in the implementation.

Reply:
Done